### PR TITLE
Add `cupy.fabs` API.

### PR DESCRIPTION
**Pull request by khushi-411**
Hi, Team!
The PR **aims** to add `cupy.fabs` API. Follows https://github.com/cupy/cupy/issues/6078#issue-1051191869.

I will be happy to know and learn from your viewpoints. Thanks!

cc: @kmaehashi

### CODE REVIEW

File: cupy/_math/misc.py [hunk 1]
```diff
@@ -201,8 +201,17 @@ def clip(a, a_min=None, a_max=None, out=None):
 absolute = _core.absolute
 
 
-# TODO(beam2d): Implement it
-# fabs
+fabs = _core.create_ufunc(
+    'cupy_fabs',
+    ('b->b', 'B->B', 'h->h', 'H->H', 'i->i', 'I->I', 'l->l', 'L->L', 'q->q',
```
Comment (khushi-411) [inline cupy/_math/misc.py:206]: The implemented dtypes are similar to `cupy.square`. Would you please let me know if more dtypes are to be added or if any types are to be removed?
Reference: [Array-protocol type strings](https://numpy.org/doc/stable/reference/arrays.dtypes.html).

Thank you! (:

Comment (toslunar) [reply cupy/_math/misc.py:206]: I suppose `['e->e', 'f->f', 'd->d']` is the choice.
```
>>> import numpy; numpy.fabs.types
['e->e', 'f->f', 'd->d', 'g->g', 'O->O']
```
and exclude `'g'` (`float128`) and `'O'` (`object`) which CuPy do not support.

File: docs/source/reference/math.rst [hunk 2]
```diff
@@ -175,6 +175,7 @@ Miscellaneous
    sign
    maximum
    minimum
+   fabs
```
Comment (toslunar) [inline docs/source/reference/math.rst:178]: Could you move the line to `absolute` and `sign`? (https://github.com/cupy/cupy/blame/v10.0.0/docs/source/reference/math.rst#L4)

Comment (khushi-411) [reply docs/source/reference/math.rst:178]: Okay, sure!

File: tests/cupy_tests/math_tests/test_misc.py [hunk 3]
```diff
@@ -161,6 +161,18 @@ def test_absolute(self):
     def test_absolute_negative(self):
         self.check_unary_negative('absolute')
 
+    @testing.for_all_dtypes(no_complex=True)
+    @testing.numpy_cupy_array_equal()
```
Comment (toslunar) [inline tests/cupy_tests/math_tests/test_misc.py:165]:
```suggestion
    @testing.numpy_cupy_allclose(atol=1e-5)
```
to align with other tests in the file.

File: tests/cupy_tests/math_tests/test_misc.py [hunk 4]
```diff
@@ -161,6 +161,18 @@ def test_absolute(self):
...
+    def test_fabs(self, xp, dtype):
+        a = xp.array([2, 3, 4], dtype=dtype)
+        return xp.fabs(a)
+
+    @testing.for_all_dtypes(no_complex=True)
+    @testing.numpy_cupy_array_equal()
```
Comment (toslunar) [inline tests/cupy_tests/math_tests/test_misc.py:171]:
```suggestion
    @testing.numpy_cupy_allclose(atol=1e-5)
```
ditto

**Review by toslunar — CHANGES_REQUESTED** [review]

File: tests/cupy_tests/math_tests/test_misc.py [hunk 5]
```diff
@@ -161,6 +161,18 @@ def test_absolute(self):
     def test_absolute_negative(self):
         self.check_unary_negative('absolute')
 
+    @testing.for_all_dtypes(no_complex=True)
+    @testing.numpy_cupy_allclose(1e-5)
```
Comment (toslunar) [inline tests/cupy_tests/math_tests/test_misc.py:165]:
```suggestion
    @testing.numpy_cupy_allclose(atol=1e-5)
```

File: tests/cupy_tests/math_tests/test_misc.py [hunk 6]
```diff
@@ -161,6 +161,18 @@ def test_absolute(self):
...
+    def test_fabs(self, xp, dtype):
+        a = xp.array([2, 3, 4], dtype=dtype)
+        return xp.fabs(a)
+
+    @testing.for_all_dtypes(no_complex=True)
+    @testing.numpy_cupy_allclose(1e-5)
```
Comment (toslunar) [inline tests/cupy_tests/math_tests/test_misc.py:171]:
```suggestion
    @testing.numpy_cupy_allclose(atol=1e-5)
```

File: tests/cupy_tests/math_tests/test_misc.py (hunk 5 above)
Comment (khushi-411) [reply tests/cupy_tests/math_tests/test_misc.py:165]: Ahh, I missed it :(

**Review by toslunar — DISMISSED** [review]

**Comment by toslunar** [comment]
How about skipping the fallback test by `@pytest.mark.skip(reason='#6282 implemented cupy.fabs')`?
https://github.com/cupy/cupy/blob/9e6fe22d8495438783621f4a73fe893764fe85b5/tests/cupyx_tests/fallback_mode_tests/test_notifications.py#L164-L182

**Comment by khushi-411** [comment]
Hi, @toslunar,
I think it will be good to skip the `test_numpy_only_pyfunc` test, as all three tests failed due to the same. Why do these tests fail?
Also, is there any other way instead of skipping these tests?

**Comment by toslunar** [comment]
> Hi, @toslunar, I think it will be good to skip the `test_numpy_only_pyfunc` test, as all three tests failed due to the same. Why do these tests fail? Also, is there any other way instead of skipping these tests?

The tests are for the feature to fallback to NumPy function.  The output of `fabs` is the same but the fallback message has gone.

My current idea to revive the tests is to `unittest.mock.patch('numpy.func_not_in_cupy', create=True)`, while it's out of scope of this `fabs` PR.

**Comment by khushi-411** [comment]
> The tests are for the feature to fallback to NumPy function. The output of fabs is the same but the fallback message has gone.

Oo, got it! Thanks, @toslunar!

> My current idea to revive the tests is to unittest.mock.patch('numpy.func_not_in_cupy', create=True), while it's out of scope of this fabs PR.

Hmm, I have to think about it a bit.

**Comment by khushi-411** [comment]
Hi, @toslunar!
A gentle ping to you to revisit the PR. I skipped the `class TestSpmm` in the `tests/cupy_tests/test_cusparse.py` tests since CI didn't pass.

Here's the output of CI: https://ci.preferred.jp/cupy.linux.cuda115/90357/

Thanks!

**Comment by khushi-411** [comment]
Hi, @toslunar, gentle ping :)

**Comment by toslunar** [comment]
> I skipped the `class TestSpmm` in the `tests/cupy_tests/test_cusparse.py` tests since CI didn't pass.

Since it passed in https://ci.preferred.jp/cupy.linux.cuda115/90347/, the failure seems a recent problem with `cupy.linux.cuda115` (fixed by #6303).
Could you exclude the last commit (https://github.com/cupy/cupy/pull/6282/commits/1a51bbe4142f99e558bb67f6432ab2fb0eff93f4) from the PR?

**Comment by khushi-411** [comment]
> Since it passed in https://ci.preferred.jp/cupy.linux.cuda115/90347/, the failure seems a recent problem with cupy.linux.cuda115 (fixed by #6303).

Oo, got it. It's interesting. Thanks for sharing the PR :)

> Could you exclude the last commit (1a51bbe) from the PR?

Yes, I did. Thanks a lot for reviewing, @toslunar!

**Review by toslunar — APPROVED** [review]

**Pull request merged**